File: angular-ssr/src/app/bike/bike.ts
interface Cart {
    cartItems: CartItem[];
    rentTotal: number;
    cartTotal: number;
    cartIsEmpty: boolean;
  }
  
  interface CartItem {
    id: number;
    bike: Bike;
    quantity: number;
  }
  
export interface Bike {
    imageSource: string;
    details: string;
    stock: number;
    price: number;
    id: number;
    model: string;
    electric: boolean;
    inStock: boolean;
  }
  

File: mfe-spring/src/main/java/com/dev/springmfe/Bike.java
package com.dev.springmfe;



import java.math.BigDecimal;

public class Bike {
    public Long id;
    public String model;
    public boolean electric;
    public String imageSource;
    public String details;
    public Integer stock;
    public BigDecimal price;
    public boolean isInStock;


    public Bike() {
    }

    public Long getId() {
        return id;
    }

    public String getModel() {
        return model;
    }

    public boolean isElectric() {
        return electric;
    }

    public String getImageSource() {
        return imageSource;
    }

    public String getDetails() {
        return details;
    }

    public Integer getStock() {
        return stock;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public boolean isInStock() {
        return isInStock;
    }
}
